File: ebs_flutter/lib/model/Payment.dart
class Payment {
  final int? id;
  final int? paymentCode;
  final int? supplierCode;
  final String? supplierName;
  final int? purchaseOrderCode;
  final double? amountPaid;
  final DateTime? payDate;
  final String? paymentMode;
  final String? paymentRef;

  Payment({
    this.id,
    this.paymentCode,
    this.supplierCode,
    this.supplierName,
    this.purchaseOrderCode,
    this.amountPaid,
    this.payDate,
    this.paymentMode,
    this.paymentRef,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: json['id'],
      paymentCode: json['paymentCode'],
      supplierCode: json['supplierCode'],
      supplierName: json['supplierName'],
      purchaseOrderCode: json['purchaseOrderCode'],
      amountPaid: json['amountPaid'] != null ? (json['amountPaid'] as num).toDouble() : null,
      payDate: json['payDate'] != null ? DateTime.parse(json['payDate']) : null,
      paymentMode: json['paymentMode'],
      paymentRef: json['paymentRef'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'paymentCode': paymentCode,
      'supplierCode': supplierCode,
      'supplierName': supplierName,
      'purchaseOrderCode': purchaseOrderCode,
      'amountPaid': amountPaid,
      'payDate': payDate?.toIso8601String(),
      'paymentMode': paymentMode,
      'paymentRef': paymentRef,
    };
  }
}

File: ebs-springboot/src/main/java/com/erp/model/Payment.java
package com.erp.model;

import javax.persistence.*;
import java.util.Date;

@Entity(name = "payment")
@Table(name = "payment")
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "payment_code")
    private Long paymentCode;

    @Column(name = "supplier_code")
    private Long supplierCode;

    @Column(name = "supplier_name")
    private String supplierName;

    @Column(name = "purchase_order_code")
    private Long purchaseOrderCode;

    @Column(name = "amount_paid")
    private Double amountPaid;

    @Column(name = "pay_date")
    @Temporal(TemporalType.DATE)
    private Date payDate;

    @Column(name = "payment_mode")
    private String paymentMode;

    @Column(name = "payment_ref")
    private String paymentRef;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Long getPaymentCode() {
		return paymentCode;
	}

	public void setPaymentCode(Long paymentCode) {
		this.paymentCode = paymentCode;
	}

	public Long getSupplierCode() {
		return supplierCode;
	}

	public void setSupplierCode(Long supplierCode) {
		this.supplierCode = supplierCode;
	}

	public String getSupplierName() {
		return supplierName;
	}

	public void setSupplierName(String supplierName) {
		this.supplierName = supplierName;
	}

	public Long getPurchaseOrderCode() {
		return purchaseOrderCode;
	}

	public void setPurchaseOrderCode(Long purchaseOrderCode) {
		this.purchaseOrderCode = purchaseOrderCode;
	}

	public Double getAmountPaid() {
		return amountPaid;
	}

	public void setAmountPaid(Double amountPaid) {
		this.amountPaid = amountPaid;
	}

	public Date getPayDate() {
		return payDate;
	}

	public void setPayDate(Date payDate) {
		this.payDate = payDate;
	}

	public String getPaymentMode() {
		return paymentMode;
	}

	public void setPaymentMode(String paymentMode) {
		this.paymentMode = paymentMode;
	}

	public String getPaymentRef() {
		return paymentRef;
	}

	public void setPaymentRef(String paymentRef) {
		this.paymentRef = paymentRef;
	}
    
}
